### PR TITLE
Add first version of IP Service proto

### PR DESCRIPTION
**Pull request by sargun**

### CODE REVIEW

**Comment by sargun** [comment]
I have docs like this:

```
title Initial Provisoning Flow

Agent->Service: Call ProvisionInstance
activate Service
Service->AWS: Create ENIs
Service->AWS: Attach ENIs
Service->Agent: Success + Cache Token
deactivate Service
```

No idea if they're valuable to include.

File: src/main/proto/netflix/titus/titus_vpc_api.proto [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    // This is the cidr Prefix length, 0-32 bits for IPv4, and 0-128 bits for IPv6.
+    uint32 prefixLength = 3;
+}
+
+message AddressAllocation {
+    string region = 1;
+    string availabilityZone = 2;
+    string subnetId = 3;
+    // address.address must be unset when requesting an allocation
+    Address address = 4;
```
Comment (andrew-leung) [inline src/main/proto/netflix/titus/titus_vpc_api.proto]: If the address is meant to be unset, should the region/zone/subnet be in a separate message passed on the request and then included in the message being returned?

File: src/main/proto/netflix/titus/titus_vpc_api.proto [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+message GCResponse {
+    repeated UsableAddress usableAddresses = 1;
+    CacheVersion cacheVersion = 2;
+}
+
+service IPService {
+    // This ProvisionInstance function has to be called at startup of the instance, and it is idempotent.
+    rpc ProvisionInstance (ProvisionInstanceRequest) returns (ProvisionInstanceResponse);
```
Comment (andrew-leung) [inline src/main/proto/netflix/titus/titus_vpc_api.proto]: I think these are internal APIs, while AllocateAddress() is intended to be called by users. Should we separate them into different services? I could envision wanting to protect the endpoints differently, plus both in the same service may confuse users.

Comment (sargun) [reply src/main/proto/netflix/titus/titus_vpc_api.proto]: Made some changes. Can you take a look?

Comment (andrew-leung) [reply src/main/proto/netflix/titus/titus_vpc_api.proto]: LGTM!

**Pull request merged**